app/src/main/java/net/squanchy/schedule/tracksfilter/widget/FilterChipView.kt

```diff
@@ -0,0 +1,239 @@
+package net.squanchy.schedule.tracksfilter.widget
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.Color
+import android.graphics.PorterDuff
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.GradientDrawable
+import android.support.annotation.ColorInt
+import android.support.annotation.Px
+import android.support.v4.content.ContextCompat
+import android.support.v4.content.res.ResourcesCompat
+import android.support.v7.widget.AppCompatTextView
+import android.util.AttributeSet
+import android.view.accessibility.AccessibilityEvent
+import android.view.accessibility.AccessibilityNodeInfo
+import android.widget.Checkable
+import net.squanchy.R
+import net.squanchy.support.graphics.darkenToEnsureTextContrasts
+import net.squanchy.support.graphics.pickBestTextColorByContrast
+
+class FilterChipView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet,
+    defStyle: Int = 0
+) : AppCompatTextView(context, attrs, defStyle), Checkable {
+
+    private var isInitialized = false
+
+    private var isChecked: Boolean = false
+    private var isBroadcastingCheckedChange: Boolean = false
+
+    private var baseColor: Int
+    private var checkedTextColorDark: Int
+    private var checkedTextColorLight: Int
+    private var uncheckedFallbackTextColor: Int
+
+    @Px
+    private var strokeWidth: Int
+
+    // It sounds weird, but a <shape> with a stroke color inflates as a GradientDrawable
+    private var backgroundDrawable: GradientDrawable
+    private var backgroundAlpha: CheckableValue<Float>
+
+    private lateinit var backgroundColor: CheckableValue<Int>
+    private lateinit var strokeColor: CheckableValue<Int>
+    private lateinit var textColor: CheckableValue<Int>
+
+    init {
+        val a = this.context.obtainStyledAttributes(attrs, R.styleable.FilterChipView, defStyle, R.style.Widget_Squanchy_FilterChipView)
+
+        checkedTextColorDark = a.getColor(R.styleable.FilterChipView_checkedTextColorDark, Color.YELLOW)
```

Comment:
Why yellow, magenta and green as defaults?

Reply:
Just because I'm still in devving mode and I want to be sure they're clearly missing if I forget them 💃

Reply:
Added a TODO a76d3c230da0178124a70c6fd47f753df040adfd